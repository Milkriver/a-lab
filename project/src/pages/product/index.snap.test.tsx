import { configureMockStore } from '@jedmao/redux-mock-store';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Product } from './index';

describe('Page: Product snapshot', () => {
  it('should render correctly', () => {
    const card = {
      id: 1,
      preview: 'http://src',
      title: 'titleCard',
      price: 200,
      availability: true,
      description: 'descriptionCard',
      images: ['http://srcImage']
    };

    const store = configureMockStore()({
      products: {
        card
      }
    });

    const { container } = render(
      <Provider store={store}>
        <Product />
      </Provider>);

    expect(container).toMatchSnapshot();
  });
});
