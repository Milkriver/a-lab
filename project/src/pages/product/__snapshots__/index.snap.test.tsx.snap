// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Product snapshot should render correctly 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="link__component_hcyko link__primary_11lt0"
      href="/"
    >
      <span
        class="link__text_hcyko"
      >
        <h1
          class="typography__component_mub95 typography__styrene-medium_g9g7d typography__bold_mub95 typography__accent_tcpuw"
        >
           A-Store
        </h1>
      </span>
    </a>
    <div
      class="menuButton"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M3 13.25h18v-2.5H3v2.5zm0-7.5v2.5h18v-2.5H3zm0 12.5h18v-2.5H3v2.5z"
        />
      </svg>
      <div
        class="gap__gap_15oeq gap__vertical_15oeq"
        data-gap-size="l"
      />
      <h1
        class="typography__component_mub95 text typography__styrene-medium_g9g7d typography__bold_mub95 typography__primary_tcpuw"
      >
        меню
      </h1>
    </div>
  </div>
  <div
    class="container"
  >
      
    <div
      class="pageWrapper"
    >
      <div
        class="grid__row_11gzm grid__component_e8mm2 grid__top_e8mm2 grid__left_e8mm2 grid__gutter-mobile-s-16_11gzm grid__gutter-desktop-m-24_11gzm"
      >
        <div
          class="grid__col_11gzm grid__component_1sygg grid__width-mobile-12_1sygg grid__width-tablet-12_1sygg grid__width-desktop-6_1sygg"
        >
          <div
            class="pageImage"
            data-testid="active-image"
            style="background-image: url(http://srcImage);"
          />
          <div
            class="gap__gap_15oeq gap__vertical_15oeq"
            data-gap-size="xl"
          />
          <div>
            <div
              class="gallery"
            />
          </div>
        </div>
        <div
          class="grid__col_11gzm grid__component_1sygg grid__width-mobile-12_1sygg grid__width-tablet-12_1sygg grid__width-desktop-6_1sygg"
        >
          <div
            class="productInfo"
          >
            <h6
              class="typography__component_mub95 typography__styrene-small_g9g7d typography__medium_mub95 typography__primary_tcpuw"
            >
              titleCard
            </h6>
            <div
              class="gap__gap_15oeq gap__vertical_15oeq"
              data-gap-size="xl"
            />
            <h6
              class="typography__component_mub95 typography__styrene-small_g9g7d typography__bold_mub95 typography__primary_tcpuw"
            >
              200
              ₽
            </h6>
            <div
              class="gap__gap_15oeq gap__vertical_15oeq"
              data-gap-size="xl"
            />
            <button
              class="button__component_s1pl2 button__m_s1pl2 button__component_17krt button__primary_17krt"
              type="button"
            >
              <span
                class="button__text_s1pl2 button__stretchText_s1pl2"
              >
                В корзину
              </span>
            </button>
            <div
              class="gap__gap_15oeq gap__vertical_15oeq"
              data-gap-size="xl"
            />
            <div
              class="description typography__primary_tcpuw typography__primary-large_11grk"
            >
              descriptionCard
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      class="copyright"
    >
      <div
        class="typography__component_mub95 text typography__styrene-xsmall_g9g7d typography__medium_mub95 typography__tertiary_tcpuw"
      >
        ©ООО "Альфа Фьюче Пипл", 2022
      </div>
    </div>
  </div>
</div>
`;
